manageiq-operator/pkg/helpers/miq-components/kafka.go

```diff
@@ -42,13 +42,34 @@ func KafkaPVC(cr *miqv1alpha1.ManageIQ) *corev1.PersistentVolumeClaim {
 		"app": cr.Spec.AppName,
 	}
 	storageReq, _ := resource.ParseQuantity(cr.Spec.KafkaVolumeCapacity)
+	if cr.Spec.StorageClassName == "" {
+		return &corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "kafka-data",
+				Namespace: cr.ObjectMeta.Namespace,
+				Labels:    labels,
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					"ReadWriteOnce",
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						"storage": storageReq,
+					},
+				},
+			},
+		}
+	}
+
 	return &corev1.PersistentVolumeClaim{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      "kafka-data",
 			Namespace: cr.ObjectMeta.Namespace,
 			Labels:    labels,
 		},
 		Spec: corev1.PersistentVolumeClaimSpec{
+			StorageClassName: cr.Spec.StorageClassName,
```

Comment:
Can we deduplicate these definitions? Maybe define the pvc then add the storage class name to the spec if it's defined in the CR?

We do something similar for the [httpd service account](https://github.com/ManageIQ/manageiq-pods/blob/3e0699ed3f4d530504f235c2593286fa2bacdb11/manageiq-operator/pkg/helpers/miq-components/httpd.go#L310) to avoid defining an entire extra deployment.

Reply:
Yes, make sense, let me update.